types/react-bootstrap-table/index.d.ts

```diff
@@ -1,66 +1,207 @@
+export type FilterCondition = 'eq' | 'like';
+
+/**
+ * Filter comparators used for NumberFilter/DateFilter filters
+ */
+export type FilterComparator = '=' | '<' | '<=' | '>' | '>=' | '!=';
+
+/**
+ * Element type to use for editing a particular column's cells.
+ */
+export type EditCellType = 'textarea' | 'select' | 'checkbox' | 'datetime';
+
+/**
+ * Position to show the Pagination Panel. Options are above the table ('top'), below the table ('bottom'), or both
+ * above and below the table ('both').
+ */
+export type PaginationPostion = 'top' | 'bottom' | 'both';
+
+/**
+ * Result type for validation when editing.
+ */
+export type EditValidatorType = 'success' | 'error';
+
+/**
+ * Used to specify the text alignment for a column.
+ */
+export type DataAlignType = 'left' | 'center' | 'right' | 'start' | 'end';
+
+/**
+ * Boostrap version number.
+ */
+export type BootstrapVersion = '3' | '4';
+
+/**
+ * CSV Field types supported.
+ */
+export type CSVFieldType = 'string' | 'number';
+
+/**
+ * Custom attributes for a column/cell/etc.
+ * Example: { 'data-attr': 'test' }
+ */
+export interface CustomAttrs {
+	[attrKey: string]: string | number | boolean;
+}
+
+/**
+ * Size per page list definition
+ */
+export type SizePerPageList = number[] | Array<{ text: string, value: number }>;
 
 /**
  * Interface spec for sepcifying functionality to handle remotely
  *
  * Consult [documentation](https://allenfang.github.io/react-bootstrap-table/docs.html#remote)
  * for more info
- *
  */
 export interface RemoteObjSpec {
-	/** If set, cell edits will be handled remotely */
+	/**
+	 * If set, cell edits will be handled remotely
+	 */
 	cellEdit?: boolean;
-	/** If set insertions will be handled remotely */
+	/**
+	 * If set insertions will be handled remotely
+	 */
 	insertRow?: boolean;
-	/** If set deletion will be handled remotely */
+	/**
+	 * If set deletion will be handled remotely
+	 */
 	dropRow?: boolean;
-	/** If set filters will be handled remotely */
+	/**
+	 * If set filters will be handled remotely
+	 */
 	filter?: boolean;
-	/** If set search will be handled remotely */
+	/**
+	 * If set search will be handled remotely
+	 */
 	search?: boolean;
-	/** If set, exporting CSV will be handled remotely */
+	/**
+	 * If set, exporting CSV will be handled remotely
+	 */
 	exportCSV?: boolean;
-	/** If set sorting will be handled remotely */
+	/**
+	 * If set sorting will be handled remotely
+	 */
 	sort?: boolean;
-	/** If set pagination will be handled remotely */
+	/**
+	 * If set pagination will be handled remotely
+	 */
 	pagination?: boolean;
 }
 
 export interface BootstrapTableProps extends Props<BootstrapTable> {
 	/**
-	 * Set version='4' to use bootstrap@4, else bootstrap@3 is used.
+	 * Bootstrap version to use, values include '3' or '4'. Defaults to '3'.
 	 */
-    version?: string;
+	version?: BootstrapVersion;
 	/**
 	 * Use data to specify the data that you want to display on table.
 	 */
-	data: any[];
+	data: Array<Row<any>>;
```

Comment:
This just becomes `any[]`, is that the intent?

Reply:
As a suggestion, consider changing the definition of `BootstrapTableProps` to something like:

```ts
export interface BootstrapTableProps<TRow extends object = any> extends Props<BootstrapTable> {
  // ...
  data: TRow[];
  // ...
}
```

The same thing goes for `FooterData`, `SelectRow`, etc.

This change wouldn't be required for PR approval, but may align more closely with your original intent.